### PR TITLE
Implement review feature

### PR DESCRIPTION
**Pull request by MarynaG**

### CODE REVIEW

File: app/views/pets/show.html.erb [hunk 1]
```diff
@@ -24,9 +24,31 @@
   <strong>photo:</strong>
     <%= cl_image_tag @pet.photo, height: 300, width: 400, crop: :fill %>
 </p>
-<%# <p>
-  <strong><%= link_to 'Check Reviews', restaurant_reviews_path(@restaurant) </strong>
-</p> %>
+
+<h3>Reviews</h3>
+<% if @pet.reviews.empty? %>
+  <p>Be the first to let a review</p>
```
Comment (Elena-Learning) [inline app/views/pets/show.html.erb]: Be the first to leave a review

File: app/views/reviews/_new.html.erb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+
```
Comment (Elena-Learning) [inline app/views/reviews/_new.html.erb]: can you put this header + form inside a div class=container to centre this on the view?

File: app/controllers/reviews_controller.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    authorize @pet
+    if (Rent.where("pet_id = ? AND user_id = ?", params[:pet_id], current_user.id).count > 0)
+      @review = Review.new(review_params)
+      @review.pet = @pet
+      authorize @review
+      if @review.save
+        redirect_to pet_path(@pet)
+      end
+    else
+      redirect_to pet_path(@pet), :notice => "You can not leave a review! You havent rentred this pet!"
```
Comment (Elena-Learning) [inline app/controllers/reviews_controller.rb]: "You can not leave a review! You have not rented this pet!"

File: config/routes.rb [hunk 4]
```diff
@@ -2,6 +2,7 @@
   devise_for :users
   resources :pets do
     resources :rents, only: [:new, :create]
+    resources :reviews, only: [:create]
```
Comment (Elena-Learning) [inline config/routes.rb]: just a question: you only nest the create of reviews (not new?)

Comment (MarynaG) [reply config/routes.rb]: We don't have a separate view for new

File: app/views/reviews/_new.html.erb (hunk 2 above)
Comment (MarynaG) [reply app/views/reviews/_new.html.erb]: Not in scope of current PR/

File: app/models/pet.rb [hunk 5]
```diff
@@ -1,6 +1,7 @@
 class Pet < ApplicationRecord
   belongs_to :user
   has_many :rents, dependent: :destroy
+  has_many :reviews, dependent: :destroy
```
Comment (0math) [inline app/models/pet.rb]: do we want to remove all reviews with the deletion of the user?

File: app/views/pets/show.html.erb (hunk 1 above)
Comment (MarynaG) [reply app/views/pets/show.html.erb]: fixed, thank you

File: app/views/pets/show.html.erb [hunk 6]
```diff
@@ -24,9 +24,31 @@
...
+<% else %>
+  <ul class="list-group">
+    <% @pet.reviews.each do |review| %>
+      <li class="list-group-item d-flex justify-content-between align-items-center">
+        <span class="badge badge-primary badge-pill">
+          <% counter = 0 %>
```
Comment (0math) [inline app/views/pets/show.html.erb]: why do you need a counter here if you use 5.times?

Comment (MarynaG) [reply app/views/pets/show.html.erb]: To display stars

File: app/models/pet.rb (hunk 5 above)
Comment (MarynaG) [reply app/models/pet.rb]: It depends on deletion of a pet.

**Pull request merged**